### PR TITLE
Allow specifying EF_PD_MCAST_LOOP via EF_VI_PD_FLAGS env var

### PR DESCRIPTION
**Pull request by vsolontsov-volant**
This PR is to let a user specifying more protection domain flags via `EF_VI_PD_FLAGS` environment variable.
Potentially it can add h/w loopback support into `solar_replay`.

### CODE REVIEW

**Review by ol-alexandra — APPROVED** [review]
The commit log should be copied from the request subject, but otherwise looks reasonable.

**Comment by vsolontsov-volant** [comment]
> The commit log should be copied from the request subject, but otherwise looks reasonable.

Thanks a lot for your review. I fixed the comment, so should be fine.
Please let me know if there's anything I can do to push it through.

File: src/lib/ciul/pd.c [hunk 1]
```diff
@@ -19,22 +19,47 @@
...
+        else if( __ef_tok_eq(s, tok_end - s, "phys") )
+            new_flags |= EF_PD_PHYS_MODE;
+        else if( __ef_tok_eq(s, tok_end - s, "default") )
+            flags = 0;
+        else if( __ef_tok_eq(s, tok_end - s, "mcast_loop") )
+            new_flags |= EF_PD_MCAST_LOOP;
+        s = tok_end + 1;
+    } while( *tok_end != '\0' );
+
+    return new_flags != 0 ? new_flags : flags;
```
Comment (rhughes-xilinx) [inline src/lib/ciul/pd.c:52]: I think this has changed the behaviour of `EF_VI_PD_FLAGS=default`

Comment (vsolontsov-volant) [reply src/lib/ciul/pd.c:52]: Not really (unless I'm missing something).
Stand-alone `default` resets the `flags` to zero (not `new_flags`).

Comment (rhughes-xilinx) [reply src/lib/ciul/pd.c:52]: Oh yes, so it does.

**Review by rhughes-xilinx — APPROVED** [review]

**Comment by ol-alexandra** [comment]
I've pushed this change to the master branch, which will be released as onload-8.0.  Eventually.

**Pull request closed without merging**